### ISSUE TITLE
QL window size is not restored on QL startup

### ISSUE THREAD

**Issue by Konzertheld** [quodlibet/quodlibet#4064]
Steps to reproduce
------------------
Start QL, maximize window, quit QL, start QL

Expected Output
---------------
QL should start maximized

Actual Output
-------------
it starts with its default window size

Test System
-----------

### Which version of Quod Libet?
4.5.0

### Which operating system 
Manjaro Gnome with Wayland

Additional Information
----------------------
I do not have this problem with other programs so I suspect it to be a problem in QL.

**Comment by declension** [comment]
Hmm, I can't replicate this on Manjaro Gnome (X11 though). Do you get any logs in `--debug`?

**Comment by declension** [comment]
Also what does ` grep maximized  ~/.quodlibet/config` say?

**Comment by Konzertheld** [comment]
> Hmm, I can't replicate this on Manjaro Gnome (X11 though). Do you get any logs in `--debug`?

Yes, it contains a lot of entries that start with `Gtk-CRITICAL` which are all duplicates of the following two lines:
```
(io.github.quodlibet.QuodLibet:10217): Gtk-CRITICAL **: 17:29:31.066: _gtk_css_image_get_concrete_size: assertion 'default_width > 0' failed
(io.github.quodlibet.QuodLibet:10217): Gtk-CRITICAL **: 17:29:31.066: _gtk_css_image_get_surface: assertion 'surface_width > 0' failed
```

> Also what does ` grep maximized ~/.quodlibet/config` say?

That file does not exist on my system. I think you meant `~/.config/quodlibet/config` for which the result is
```browser_searchbar_maximized = 0
   browser_tracklist_maximized = 0
   plugin_prefs_maximized = 0
   quodlibet_information_maximized = 0
   quodlibet_maximized = 4
   quodlibet_properties_maximized = 0
```
when I quit QL in maximized state and the same with `quodlibet_maximized = 0` when I quit in the state in which it starts.

**Comment by 0rzech** [comment]
It's the same for me. The repro in my case is to tile QL to the left or right side of the screen in Gnome Shell. After that, QL is unable to start maximized again even after I maximize the window back.

I have diffed working `config` with the one that does not work and it seems that `quodlibet_size_` is what makes QuodLibet unable to start maximized.

While the value is changed when I make QL window smaller, eq. by tiling to the left or right, it is not changed back when I maximize QL.

**Comment by glaubersm** [comment]
Can reproduce with Quod Libet 4.5.0 on Arch Linux.

**Comment by toto789** [comment]
Can reproduce with Quod Libet 4.6.0 on Ubuntu 23.10.

**Comment by Saroumane** [comment]
On KDE wayland (archlinux), I tried @ksdrummer  method : on next startup quodlibet (4.6.0) used half left of the screen, the maximized state is still not saved.

Disclaimer  : I use a 3840x2160 resolution with 200% scaling 
Switching to 3840x2160 / 100% scaling does not help, **but as soon as I switch to 1920x1080 / 100% scale, the problem is gone.**
My theory : the problem is triggered by any resolution higher than 1920x1080 (at least  with wayland)

**The workaround I used :** 

I spotted this value `~/.config/quodlibet/config `
quodlibet_size_ = 959 992

I edited (doubled) the 1st value : 
quodlibet_size_ = 1918 992

And voilà, quodlibet is virtually maximized, using all screen available (even if the icon window in top right corner does not reflect this "maximized" state)

**Comment by T3HM3T4L** [comment]
I'm still having this issue on KDE Plasma, Endeavour OS (Arch Linux) Wayland.

You can indeed come up with a combination of outlining the window (meta + arrrow key) and then making the window bigger manually to have a pseudo-maximized state which QL will remember, but this bug still hasn't been fixed in 3 years.

I have also tried using KDE's Window Rules to tell QL to maximize the main screen on start-up, but I was unable to do that for just the main screen: either all QL screens (including song information etc) would be maximized or the maximization wouldn't take, even if I made specifications.
